pkg/controllers/machine_controller.go

```diff
@@ -154,12 +169,54 @@ func (r *MachineReconciler) Start(ctx context.Context) error {
 		}
 	}()
 
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		sizeLogger := r.log.WithName("size")
+		defer wg.Done()
+
+		wait.UntilWithContext(ctx, func(ctx context.Context) {
```

Comment:
anonymous function can be defined as standard function separately.

Reply:
Will keep it like this, since otherwise the logger needs to be created on every function call.

Reply:
you have 2 options:

- use combination of anonymous function with standard function and use closure principle for arguments.
- don't use wait but only simple two for loops.

And pls stop resolving conversation, i am author of conversation and i have to be ok with your fixes or arguments. Resolve conversation of someone else has to be edge case (sick, doesn't react, ...).

It is extremely hard check what you fixed or what you just closed. Especially when one conversion affects more conversation i cannot use outdated label and i have to open every conversantion.

Reply:
As already mentioned, I will keep it like this since it dosen't bring any benefits.

Reply:
you create spaghetti code and from ~100 lines function is 40 lines of code independent logic. If you create separate function as checkingVolumeSizeAsRequeue, it will be more readable and people don't have to study of code for know what it is doing.